midx.c

```diff
@@ -1014,6 +1014,9 @@ int verify_midx_file(const char *object_dir)
 	for (i = 0; i < m->num_packs; i++) {
 		if (prepare_midx_pack(m, i))
 			midx_report("failed to load pack in position %d", i);
+
+		if (m->packs[i])
+			install_packed_git(the_repository, m->packs[i]);
```

Comment:
I am slightly worried that this might result in `close_one_pack()` "pulling out the data under midx' rug". IOW I could imagine that there are fields in `m` that point into that `m->packs` array, and if (part of) that gets released, those pointers go stale.

Reply:
Maybe run the `midx` tests under `--valgrind`, to see whether this triggers? But then, we might not even get to `close_one_pack()` in the regression tests' toy examples...

Reply:
I'm quite new to the packfile code, but it looks like close_one_pack() doesn't actually change any of the allocated objects -- it just closes the fd of one of them and leaves the packed_git structure in place. This lets them re-open the packfile if they later need it again.

Reply:
Ooooh, I totally missed that `close_one_pack()` will not `free()` the `struct packed_git *`, but it will only close the file descriptor within. So I withdraw my concern here.